LayoutTests/http/tests/site-isolation/resources/dark-mode-background.html

```diff
@@ -0,0 +1,19 @@
+<style>
+    @media (prefers-color-scheme: dark) {
+        body {
+            background-color: green;
+        }
+    }
+    @media (prefers-color-scheme: light) {
+        body {
+            background-color: red;
+        }
+    }
+</style>
+<body>
+    <script>
+        if (window.internals)
+            internals.settings.setUseDarkAppearance(true);
```

Comment:
Does this test fail without the code change?

It seems like `internals.setting.setUseDarkAppearance` only sets the value on the `Page` instance in the current process, and doesn't do anything to notify the UIProcess (and thus other processes).

It might be better to have a test where the setter is in the outer document, but the relevant `@media` rule remains in the cross-origin iframe.

I think we'd need a new `internals` API that sets dark appearance via an IPC message to the `WebPageProxy`, and probably a promise return value so the test can wait for remote processes to be notified.

Reply:
> Does this test fail without the code change?
> 

Yes it was failing because I was calling `internals.settings.setUseDarkAppearance(true);` in the cross-origin frame page. So `Page::m_useDarkAppearance` of the remote frame was set to true. But `Page::useDarkAppearance()` returns false always for remote frame without my fix.

> It seems like `internals.setting.setUseDarkAppearance` only sets the value on the `Page` instance in the current process, and doesn't do anything to notify the UIProcess (and thus other processes).
> 
> It might be better to have a test where the setter is in the outer document, but the relevant `@media` rule remains in the cross-origin iframe.
> 

Done. I moved setting the dark appearance to the outer document to verify PageProxy sends the message to all the WebContentProcesses.

> I think we'd need a new `internals` API that sets dark appearance via an IPC message to the `WebPageProxy`, and probably a promise return value so the test can wait for remote processes to be notified.

I added the new API TestRunner::setUseDarkAppearanceForTesting() since TestRunner is the UIProcess for running WKR.